forward/forward.go

```diff
@@ -160,6 +160,9 @@ func (f *Forwarder) cloneRequest(req *http.Request, u *url.URL) (*http.Request,
 	// Request Header
 	outReq.Header = make(http.Header)
 	copyHeadersForForwarding(outReq.Header, req.Header)
+	// Ensure we have a HOST header (important for Go 1.6+ because http.Server
+	// strips the HOST header from the inbound request)
+	outReq.Header.Set("Host", req.Host)
```

Comment:
This is not immediately necessary while we're building with Go 1.5, but if we ever switch to Go 1.6 it will be.